src/client.ts

```diff
@@ -103,23 +102,36 @@ export class APIClient {
     for (const user of users) {
       await iteratee({
         ...user,
-        employeeDetails: employees[user.employeeId] || {},
+        employeeDetails: employees.get(user.employeeId.toString()) || {},
       });
     }
   }
 
-  private async fetchEmployeeDirectory(): Promise<BambooHREmployeesMap> {
+  public async iterateEmployees(
+    iteratee: ResourceIteratee<BambooHREmployee>,
+  ): Promise<void> {
+    const employees = await this.fetchEmployeeDirectory();
```

Comment:
I missed this earlier, but it looks like `await this.fetchEmployeeDirectory();` is being called during `iterateEmployees` and `iterateUsers`. What do you think about caching that data? Is the complete set of employees in a single request to `'v1/employees/directory'`?